packages/bierzo-wallet/src/communication/extension.ts

```diff
@@ -0,0 +1,45 @@
+import { ChainId, Identity } from "@iov/bcp";
+
+import { sendGetIdentitiesRequest } from "./identities";
+
+export interface ExtensionStatus {
+  readonly connected: boolean;
+  readonly installed: boolean;
+  readonly identities: ReadonlyMap<ChainId, Identity>;
+}
+
+/**
+ * Groups the identites by blockchain. Returns the first identity of each chain.
+ *
+ * The browser extension is supposed to only send one identity per chain.
+ * @see https://github.com/iov-one/ponferrada/issues/447
+ */
+export function groupIdentitiesByChain(identities: readonly Identity[]): ReadonlyMap<ChainId, Identity> {
+  const out = new Map<ChainId, Identity>();
+  for (const identity of identities) {
+    if (!out.has(identity.chainId)) {
+      out.set(identity.chainId, identity);
```

Comment:
You can `set` new value without checking for it's existence.

Reply:
Right, but this overrides existing entries. In this function we always want to use the first entry for any given chain ID. Because of that, if the chain ID exists already we don't override.

Reply:
thanks